src/http.c

```diff
@@ -106,22 +109,40 @@ void http_update(struct http *http,
                  const void *data,
                  unsigned int data_len,
                  unsigned int report_http) {
+
+    struct http_message *message = NULL;
+    char raw_header[HTTP_MAX_LEN] = {0};
+    unsigned int tmp_len = 0;
+    int rc = 0;
+
     if (!report_http || data_len == 0) {
         return;
     }
+
+    if (http == NULL) {
+        return;
+    }
+
+    if (http->num_messages >= HTTP_MAX_MESSAGES - 1) {
+        /* Already at maximum message capacity */
+        return;
+    }
+
+    /* Get the current message datastore */
+    message = &http->messages[http->num_messages];
   
-    if (http->header == NULL) {
-        unsigned int len = (data_len + MAGIC) < HTTP_LEN ? (data_len + MAGIC) : HTTP_LEN;
-        /*
-         * note: we leave room for null termination in the data buffer
-         */
-       
-        http->header = malloc(len);
-        if (http->header == NULL) {
-            return; 
-        }
-        memset(http->header, 0x0, len);
-        http->header_length = memcpy_up_to_crlfcrlf_plus_magic(http->header, data, len);
+    /*
+     * Leave room for null termination in the data buffer
+     */
+    tmp_len = (data_len + MAGIC) < HTTP_MAX_LEN ? (data_len + MAGIC) : HTTP_MAX_LEN;
+
+    memcpy_up_to_crlfcrlf_plus_magic(raw_header, data, tmp_len);
```

Comment:
This function returns a value i the current length. Does this function need to return the value?

Reply:
I could add a check here that if i == 0, then we will be unable to successfully call http_parse_message. So bail early here